src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    private final Map<Class<? extends Machine>, MachineProducer<? extends Machine>> producerMap
+            = new HashMap<>();
+
+    {
+        producerMap.put(Truck.class, new TruckProducer());
+        producerMap.put(Bulldozer.class, new BulldozerProducer());
+        producerMap.put(Excavator.class, new ExcavatorProducer());
+    }
+
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        MachineProducer<? extends Machine> producer = producerMap.get(type);
+        if (producer != null) {
+            return (List<T>) producer.get();
```

Comment:
The cast `(List<T>) producer.get()` is unchecked and could lead to a `ClassCastException` at runtime if the producer does not return a list of the expected type `T`. Consider adding a check or using a safer approach to ensure type safety.